### ISSUE TITLE
False positive in detectSuBinaries function

### ISSUE THREAD

**Issue by sangcx5** [aimardcr/AndroidNativeGuard#10]
The detectSuBinaries function check a binary exists by make a call to SecureAPI::openat(AT_FDCWD, suBinary, O_RDONLY, 0)

```C++
bool RootDetect::detectSuBinaries() {
    for (const char *suBinary : suBinaries) {
        int fd = SecureAPI::openat(AT_FDCWD, suBinary, O_RDONLY, 0);
        if (fd < 0) {
            LOGI("RootDetect::execute su binary detected: %s", suBinary);
            SecureAPI::close(fd);
            return true;
        }
       ...
}
```

But with a file that not exist, `SecureAPI::openat(AT_FDCWD, suBinary, O_RDONLY, 0)` return -2
so if device doesn't have any file in this list, your code will indicate that the device is rooted
```C++
static const char *suBinaries[] = {
    "/data/local/su",
    "/data/local/bin/su",
    "/data/local/xbin/su",
    "/sbin/su",
    "/su/bin/su",
    "/system/bin/su",
    "/system/bin/.ext/su",
    "/system/bin/failsafe/su",
    "/system/sd/xbin/su",
    "/system/usr/we-need-root/su",
    "/system/xbin/su",
    "/cache/su",
    "/data/su",
    "/dev/su"
};
```
So please recheck and update root detection logic
POC:
![image](https://github.com/user-attachments/assets/01124b0d-5d3c-4beb-aa5c-e41f66053a5a)

**Comment by aimardcr** [comment]
Hi, this is a known issue and will be updated as soon as possible.

**Comment by sangcx5** [comment]
> Hi, this is a known issue and will be updated as soon as possible.

ah okay, thank you

**Comment by priyaji1** [comment]
@aimardcr Does false positive in detectSuBinaries function is fixed as i am using the code but it is saying a non rooted device as rooted , can you confirm one more thing that this rooted check will work when app is in Zygisk denylist?

**Comment by aimardcr** [comment]
HI, the false positive is WIP. I haven't tried myself on the Zygisk denylist.

**Comment by priyaji1** [comment]
Can you please let me know how I could detect that my app is in zygisk
denylist.

**Comment by aimardcr** [comment]
Fixed in ef5c80a0a7f80380c3befa5581af361c3871d06e.

**Comment by priyaji1** [comment]
Thank you so much @aimar so you have added check for zygisk denylist also?
